app/api/v2/handlers/ability_api.py

```diff
@@ -22,35 +22,47 @@ def add_routes(self, app: web.Application):
+                          descriptions='Provides a list of all available abilities.')
     @aiohttp_apispec.querystring_schema(BaseGetAllQuerySchema)
     @aiohttp_apispec.response_schema(AbilitySchema(many=True, partial=True))
     async def get_abilities(self, request: web.Request):
         abilities = await self.get_all_objects(request)
         return web.json_response(abilities)
 
-    @aiohttp_apispec.docs(tags=['abilities'])
+    @aiohttp_apispec.docs(tags=['abilities'], summary='Get an ability.',
+                          description='Provides one ability based on its ability id.')
     @aiohttp_apispec.querystring_schema(BaseGetOneQuerySchema)
     @aiohttp_apispec.response_schema(AbilitySchema(partial=True))
     async def get_ability_by_id(self, request: web.Request):
         ability = await self.get_object(request)
         return web.json_response(ability)
 
-    @aiohttp_apispec.docs(tags=['abilities'], summary='"name", "tactic", and "executors" are all required fields.')
+    @aiohttp_apispec.docs(tags=['abilities'], summary='Creates a new ability.',
+                          description='Creates a new adversary based on the `AbilitySchema`. '
+                                      '"name", "tactic", and "executors" are all required fields.')
     @aiohttp_apispec.request_schema(AbilitySchema)
     @aiohttp_apispec.response_schema(AbilitySchema)
     async def create_ability(self, request: web.Request):
         ability = await self.create_on_disk_object(request)
         return web.json_response(ability.display)
 
-    @aiohttp_apispec.docs(tags=['abilities'], summary='"name", "tactic", and "executors" are all required fields.')
+    @aiohttp_apispec.docs(tags=['abilities'], summary='Creates new or updates an existing ability.',
+                          description='Given an ability based on the `AbilitySchema` if the identifier '
+                                      'matches an existing ability it will update it replacing it with '
+                                      'the contents received on the request. Otherwise a new ability will '
+                                      'be created. "name", "tactic", and "executors" are all required fields.')
     @aiohttp_apispec.request_schema(AbilitySchema(partial=True))
     @aiohttp_apispec.response_schema(AbilitySchema)
     async def create_or_update_ability(self, request: web.Request):
         ability = await self.create_or_update_on_disk_object(request)
         return web.json_response(ability.display)
 
-    @aiohttp_apispec.docs(tags=['abilities'])
+    @aiohttp_apispec.docs(tags=['abilities'], summary='Updates an existing ability.',
+                          description='Given an ability based on the `AbilitySchema` if the identifier '
+                                      'matches an existing ability it will update it replacing it with '
+                                      'the contents received on the request. "name", "tactic", and '
```

Comment:
This is a little bit nitpicky, but the wording might be a little confusing here because we treat PATCH requests in the API as updates, whereas PUT requests are object replacement (even though the endpoint is called create_or_update). Could you change this to something like: "Updates an ability based on the `AbilitySchema` values provided in the message body."

`name`, `tactic`, and `executors` are not required for this endpoint, because users don't have to submit an entire object for creation - they can just update select fields (excluding `ability_id`, `requirements`, and `additional_info`)

Reply:
I believe that because the `"name", "tactic" and "executors" required` line was already present when I started working on this was already there I left them, but good that you noticed.